service Service {
    rpc GetLastExaminationByPatient(Patient) returns (Examination) {}
    rpc getAllExaminations(Empty) returns (stream Examination) {}
    rpc getAllExaminationWithGivenParameter(Parameter) returns (stream Examination) {}
    rpc insertExamination(Examination) returns (Empty) {}
}

message Examination {
    required int32 id = 1;
    optional string date = 2;
    optional Doctor doctor = 3;
    optional Results results = 4;
}

// e.g WhiteBloodCells -> "20.5"
// e.g RedBloodCells -> "30"
message Results {
     repeated Parameter parameters = 1;
}

message Parameter {
    required string parameter_name = 1;
    required string value = 2;
}

message Patient {
    required string first_name = 1;
    required string last_name = 2;
    repeated Examination examinations = 3;
}

message Doctor {
    required string first_name = 1;
    required string last_name = 2;
    required string specialization = 3;
}

message Empty {
}